### DISCUSSION TITLE
Converting some MET config items to METplus

### DISCUSSION THREAD

**Discussion by PerryShafran-NOAA** [dtcenter/METplus#1520] [Configuration] [answered]
Hi, there,

A colleague of mine has some HYSPLIT (dust) verification working in MET but he never got around to converting to METplus, so I'm doing it. I wanted to check in on converting the configurations.

Here are the configurations from the original MET config file:

fcst = {
   wind_thresh = [ NA ];

   field = [
      {
        name       = "LIPMF";
        convert(LIPMF) = 10^LIPMF;
        level      = [ "Z0-5000" ];
        lead_time  = "${CUR_LEAD_HOUR}";
        censor_thresh    = [ <${SMALLNUMBER} ];
        censor_val       = [ 0. ];
        cat_thresh = [ <1.0,  >1.0, >2.0, >5.0, >10.0, >15.0, >20.0 ];
      }
   ];

};
obs = {
   wind_thresh = [ NA ];

   field = [
      {
        name       = "DEN";
        convert(DEN) = DEN * 10^9;
                level      = [ "Z5000" ];
                censor_thresh    = [ >-1.5&&<0., >=-3.5&&<=-1.5, <-3.5 ];
        censor_val       = [ -9999,     0.000001,      -9999 ];
        cat_thresh = [ <1.0,  >1.0, >2.0, >5.0, >10.0, >15.0, >20.0 ];
      }
   ];

Here is what I have thus far for the METplus configurations:

FCST_VAR1_NAME = LIMPF
FCST_VAR1_OPTIONS = convert(x) = 10^x;
FCST_VAR1_LEVEL = Z0-5000
FCST_VAR1_OPTIONS = censor_thresh = ltSMALLNUMBER; censor_val = 0.;
FCST_VAR1_THRESH = <1.0,  >1.0, >2.0, >5.0, >10.0, >15.0, >20.0
OBS_VAR1_NAME = DEN
OBS_VAR1_OPTIONS = convert(x) = x * 10^9;
OBS_VAR1_LEVEL = Z5000
OBS_VAR1_THRESH = <1.0,  >1.0, >2.0, >5.0, >10.0, >15.0, >20.0

(assume SMALLNUMBER is an actual number like -9999.0 or something

The sticking point for me thus far is I can do a censor_thresh and a censor_val, but what do I do when I have three of them like  in the obs field?

Thanks for any assistance you can provide.

Perry

**Comment by j-opatz** [comment, marked as answer]
Hi Perry,

And thank you for your question. You've done a great job at transferring the MET configuration options over to the METplus configuration file. To your question, you should be OK to pass in the observation field censor thresholds and values the same way you did with the forecast field:list them inside of brackets,  in the order they correspond to each other, with commas between each value:

`OBS_VAR1_OPTIONS = censor_thresh = [ >-1.5&&<0., >=-3.5&&<=-1.5, <-3.5]; censor_val = [ -9999, 0.000001, -9999 ];` 

The substitution logic in METplus should correctly interpret the above line the same way as the MET configuration option you provided. I ran a quick test on a similarly complex censor_thresh/censor_val and it ran successfully.

Please let us know if there are any other problems or questions that stem from this conversion process and we'll do our best to help you out.

**Reply by PerryShafran-NOAA** [reply]
Thanks for checking my work and giving the advice about the options line!  I'll give that a shot and hopefully it will work.  Will take a bit to complete configuration and testing.  Thanks so much!

BTW, my colleague had a lead_time = "${CUR_LEAD_HOUR}"; in his MET configuration file, which I think is just the forecast lead.  This isn't something that goes in the FCST or OBS options - the lead_time option?  

Perry

**Reply by georgemccabe** [reply]
Hi Perry,

lead_time is a keyword for a fcst or obs field setting.

From https://met.readthedocs.io/en/latest/Users_Guide/config_options.html#fcst:

> The “lead_time” entry specifies the lead time in HH[MMSS] format. This entry can be included in the “fcst” entry as shown below or included in the “field” entry if the user would like to use different lead times for different fields.

This can be set using FCST_VAR\<n\>_OPTIONS or OBS_VAR\<n\>_OPTIONS.

**Comment by PerryShafran-NOAA** [comment]
Thanks, I was able to complete the configuration file and generate stat files for this case.
